statchatui/src/Compare/Compare.jsx

```diff
@@ -21,10 +21,53 @@ const Compare = ({ signedInUserData }) => {
   const [isLoading, setIsLoading] = useState(false);
   const [otherUserStats, setOtherUserStats] = useState(null);
   const [signedInUserStats, setSignedInUserStats] = useState(null);
+  const [userQueue, setUserQueue] = useState([]);
+  const [showStats, setShowStats] = useState(null);
+  const storedUserQueue = sessionStorage.getItem("userQueue");
+
+  //get the previously stored userqueue everytime the page mounts
+  useEffect(() => {
+    if (storedUserQueue) {
+      setUserQueue(JSON.parse(storedUserQueue));
+    }
+  }, []);
+
+  //everytime userqueue is updated, sessionStorage will be updated with the new value.
+  useEffect(() => {
+    sessionStorage.setItem("userQueue", JSON.stringify(userQueue));
+  }, [userQueue]);
```

Comment:
This works, but consider using fewer `useEffect`s because it is pretty confusing to read (you have to jump around to different places in the code to figure out what happens when). 

You're basically updating user queue in two places: 
1. On initialization
2. When the user submits the form.

So it'd be easy to just call this code in those places!

Reply:
i deleted the first useeffect but this one changes it everytime it updates which is what i want

Reply:
i only use one useeffect now